src/Microsoft.AspNet.Razor/Parser/TagHelpers/TagHelperParseTreeRewriter.cs

```diff
@@ -267,6 +280,64 @@ private IEnumerable<string> GetAttributeNames(Block tagBlock)
             return attributeNames;
         }
 
+        private void ValidateParentTagHelperAllowsContent(SyntaxTreeNode child, ErrorSink errorSink)
+        {
+            var allowedChildren = _currentTagHelperTracker?.AllowedChildren;
+            if (allowedChildren != null)
+            {
+                var childSpan = (Span)child;
```

Comment:
Unconditional cast looks dangerous. Either pass in `Span` (i.e. confirm type in caller) or check `IsBlock` and use `FindFirstDescendentSpan((Block)child)` if that's `true`.